File: src/evai/command/EventCommands.java
package evai.command;

import arc.*;
import mindustry.game.*;
import evai.monitor.*;
import java.util.*;

public class EventCommands implements CommandRegistry.CommandHandler {
    @Override
    public CommandResult handle(String action, String[] args) {
        switch (action) {
            case "listen": return listen(args);
            case "fire": return fire(args);
            default: return new CommandResult(false, "unknown event command: " + action);
        }
    }

    private CommandResult listen(String[] args) {
        if (args.length == 0) return new CommandResult(false, "usage: listen <event>/all/stop/list");
        switch (args[0].toLowerCase()) {
            case "all": {
                int count = EventMonitor.listenAll();
                return new CommandResult(true, "Listening to " + count + " events");
            }
            case "stop": {
                if (args.length > 1) {
                    EventMonitor.stop(args[1]);
                    return new CommandResult(true, "Stopped listening to " + args[1]);
                }
                EventMonitor.stopAll();
                return new CommandResult(true, "Stopped all listeners");
            }
            case "list": {
                List<String> events = EventMonitor.list();
                Map<String, Object> data = new HashMap<>();
                data.put("listening", events);
                return new CommandResult(true, "", data);
            }
            default: {
                boolean ok = EventMonitor.listen(args[0]);
                return ok ? new CommandResult(true, "Listening to " + args[0]) : new CommandResult(false, "Unknown event: " + args[0]);
            }
        }
    }

    private CommandResult fire(String[] args) {
        if (args.length == 0) return new CommandResult(false, "usage: fire <event>");
        String name = args[0];
        try {
            for (Class<?> clazz : EventType.class.getClasses()) {
                if (clazz.getSimpleName().equals(name + "Event")) {
                    Object instance = clazz.getDeclaredConstructor().newInstance();
                    Events.fire(instance);
                    return new CommandResult(true, "Fired " + name + "Event");
                }
            }
            return new CommandResult(false, "Event not found: " + name + "Event");
        } catch (Exception e) {
            return new CommandResult(false, "Error: " + e.getMessage());
        }
    }
}

File: src/evai/command/EventCommands.kt
package evai.command

import arc.*
import mindustry.game.*
import evai.command.CommandRegistry.CommandHandler
import evai.monitor.*

class EventCommands : CommandHandler {
    override fun handle(action: String, args: Array<String>): CommandResult = when (action) {
        "listen" -> {
            if (args.isEmpty()) return CommandResult(false, "usage: listen <event>/all/stop/list")
            when (args[0].lowercase()) {
                "all" -> {
                    val count = EventMonitor.listenAll()
                    CommandResult(true, "Listening to all $count events")
                }
                "stop" -> {
                    if (args.size > 1) {
                        EventMonitor.stop(args[1])
                        CommandResult(true, "Stopped listening to ${args[1]}")
                    } else {
                        EventMonitor.stopAll()
                        CommandResult(true, "Stopped all listeners")
                    }
                }
                "list" -> {
                    val events = EventMonitor.list()
                    CommandResult(true, data = mapOf("listening" to events))
                }
                else -> {
                    EventMonitor.listen(args[0])
                    CommandResult(true, "Listening to ${args[0]}Event")
                }
            }
        }
        "fire" -> {
            if (args.isEmpty()) return CommandResult(false, "usage: fire <event> [args]")
            val eventName = args[0]
            val eventArgs = args.drop(1).toTypedArray()
            try {
                val eventClass = EventType::class.java.classes.find { it.simpleName == "${eventName}Event" }
                    ?: return CommandResult(false, "event not found: ${eventName}Event")
                val instance = eventClass.getDeclaredConstructor().newInstance()
                Events.fire(instance)
                CommandResult(true, "Fired ${eventName}Event")
            } catch (e: Exception) {
                CommandResult(false, "error firing event: ${e.message}")
            }
        }
        else -> CommandResult(false, "unknown event command: $action")
    }
}